uiowa/brand-icon-browser
# Scroll-to-top behavior when changing categories or clicking search terms

Resolves #62 

This PR ensures that after a route change, the user's browser scrolls to the top of the page. I originally scoped this to clicking categories or tags specifically, but this PR should now cover all route changes including categories and tags, and it should be future-proof for additional routes or functionality that may be added later.

## How to test
Here are some specific test scenarios to ensure the scroll to top behavior is working. It's a bit tricky to test because some categories don't have enough icons to fill the page entirely and it will seem like the browser is scrolling to the top and working fine, but this is the browser's default behavior when a page isn't filled out entirely.

### Scenario A: 
1. Check out this branch
2. run `npm run serve` 
3. Click on a category that has enough icons to necessitate scrolling (e.g., "Patient Care")
4. Scroll down through the icons a bit
5. Click the "All brand icons" category
6. Observe the scroll behavior: the browser should scroll to the top

### Scenario B:
1. Follow Steps 1-2 in Scenario A
2. Stay in "All Brand Icons" category
3. Scroll down quite a bit
4. Find a health care-related icon and select it
5. Click the "#patient care" tag that's associated with it
6. Observe the scroll behavior: the browser should scroll to the top
